src/thor/trace_route_action.cc

```diff
@@ -400,10 +400,15 @@ void thor_worker_t::build_route(
       bool uturn_off_of =
           next_segment && segment->edgeid != next_segment->edgeid && segment->target < 1.f;
       if (uturn_onto || uturn_off_of) {
```

Comment:
seems like maybe we should  handle each uturn case separately and not modify both at the same time... @yuzheyan and i just recently wrote this..

Reply:
Not sure what you mean here? Another question is whether Uturns always cause a discontinuity? A couple test cases I have it seems to create a discontinuity and multiple routes.

Reply:
sorry i havent had time to look at this. what i mean is that we should handle both onto and off_of in the same if. but rather have each one in its own if where it modifies only one side of the pair in the discontinuities list.

in this case yes, they should always. the boolean above says you didnt finish the edge which means you either be at the beginning or end of a route or have made a uturn in the "middle" of an edge.

Reply:
this just bit me. ive implemented the approach i mentioned above and will pr it shortly

Reply:
#2517